ext/standard/var_unserializer.re

```diff
@@ -1243,37 +1241,39 @@ object ":" uiv ":" ["]	{
 		}
 
 		/* Check for unserialize callback */
-		if ((PG(unserialize_callback_func) == NULL) || (PG(unserialize_callback_func)[0] == '\0')) {
+		if ((PG(unserialize_callback_func) == NULL) || (ZSTR_LEN(PG(unserialize_callback_func)) == 0)) {
 			incomplete_class = 1;
 			ce = PHP_IC_ENTRY;
 			break;
 		}
 
-		/* Call unserialize callback */
-		ZVAL_STRING(&user_func, PG(unserialize_callback_func));
+        /* Find unserialize callback */
+        zend_function *callback_fn = zend_hash_find_ptr_lc(EG(function_table), PG(unserialize_callback_func));
```

Comment:
This breaks some valid callback names, such as:

 - `\foo` (leading `\`)
 - `Foo::bar` (static method)

Reply:
Ah I didn't think of static methods, I'll add some tests to ensure this doesn't get missed. And I'll think about a solution if there is a good one.

Reply:
I didn't understand why you replaced call_user_function() with a manual function lookup + zend_call_known_function. Is there a particular reason?

Reply:
I am trying to see if it is possible to get rid of the `function_name` field of the FCI struct, and there are not many uses of `call_user_function()` (currently 29) on top of it having the now useless `symbol_table` argument (yes I know it's a macro so it doesn't matter).